lib/msf/core/exploit/dns/enumeration.rb

```diff
@@ -62,7 +62,7 @@ def dns_axfr(domain)
       end
       next if zone.blank?
       records << zone
-      print_good("#{domain} Zone Transfer: #{zone}")
+      print_good("#{domain} Zone Transfer: #{zone.to_s[1...-1]}") # Use [1...-1] to remove the stray [ and ] characters from the output.
```

Comment:
Since zone is actually an array of `Net::DNS::Packet`, this whole section should probably be updated accordingly. It looks like it used to expect `zone` to be a string and not an array of record answers.

```suggestion
      next if zone.empty?
      records += zone
      print_good("#{domain} Zone Transfer: #{zone.map(&:inspect).join}")
```
Specifically note:
* Using `#empty?` instead of blank because we're expecting an array and not a string
* Concatenating the array with `records += zone` instead of creating a multi-dimensional array
* Explicitly converting each packet to a string using `Net::DNS::Packet#inspect` which looks like the intended API

I checked and this function is only defined here and only used by `auxiliary/gather/enum_dns` so it should be fine to fix the multi-dimensional array return value.

Reply:
Thanks, this makes a lot of sense! I'll see if I can't get this PR updated with your suggested changes 👍

Reply:
Woops forgot to make this edit let me go ahead and do that so long.

Reply:
Edits made in https://github.com/rapid7/metasploit-framework/pull/14250/commits/ab7ed90457da790cdc09e1708409d832e7e6701b, and had a lengthy discussion over whether or not to use `.empty?` or `.blank?`. Ultimately decided to use `.blank?` as it uses both `.empty?` anyway and it also does a `nil` check on the object.

With that being said I do agree with @smcintyre-r7's point that he raised that `.blank?` on an array doesn't really make much sense from an explanation point of view, as you don't exactly say an array object is blank, you say its empty.

That being said as this library is using the `.blank?` call in a number of places already I'm going to follow this library's convention and use `.blank?` here purely so that the code doesn't become a jumbled mess of different approaches and also because it implements more checks in one line.

If someone does wish to change this down the line in the future then they are welcome to do it in a separate PR and I'd be happy to assist 👍